src/repository.py

```diff
@@ -48,6 +96,92 @@ def __init__(self, repository: Repo, github_repository: Repository) -> None:
         self._github_repo = github_repository
         self._configure_git_user()
 
+    @cached_property
+    def base_path(self) -> Path:
+        return Path(self._git_repo.working_dir)
+
+    @property
+    def metadata(self) -> Metadata:
+        return get_metadata(self.base_path)
+
+    @property
+    def has_docs_directory(self) -> bool:
+        return has_docs_directory(self.base_path)
+
+    @property
+    def current_branch(self) -> str:
+        return self._git_repo.active_branch.name
+
+    @contextmanager
+    def with_branch(self, branch_name: str) -> 'Client':
```

Comment:
Is the quote needed in 3.11?

Reply:
Uhm, my IDE was complaining, but maybe it would work anyway...not sure... I'll try this out, but here the problem here is that the function should return the same type of the class this method belongs to. Was this somehow overcome by newer versions of python/mypy?

Reply:
I can't remember whether they fixed that, I think actually it was supposed to be but then delayed because of some problems in maybe fast API? I'll resolve the comment for now